CRM/Profile/Form/Edit.php

```diff
@@ -98,8 +98,9 @@ public function preProcess() {
 
         if ($id != $userID) {
           // do not allow edit for anon users in joomla frontend, CRM-4668, unless u have checksum CRM-5228
+          // see also CRM-19079 for modifications to the condition
           $config = CRM_Core_Config::singleton();
-          if ($config->userFrameworkFrontend) {
+          if ($config->userFrameworkFrontend && $config->userSystem->is_joomla) {
```

Comment:
I have a general preference for calling overridable functions on the class - ie if ($config->userSystem->supportsAnonymousProfileUse() )

I'm not sure the exact name of what the function should be but I prefer to refer to a feature or capability of the CMS rather than the CMS itself - which is more flexible & the comment block on the CMS becomes a good place to explain what is going on (rather than those couple of lines of comments being in the form class)